src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1864,37 +1871,23 @@ void TorrentImpl::handleFileRenamedAlert(const lt::file_renamed_alert *p)
     // Remove empty leftover folders
     // For example renaming "a/b/c" to "d/b/c", then folders "a/b" and "a" will
     // be removed if they are empty
-    const QString oldFilePath = m_filePaths.at(fileIndex);
-    const QString newFilePath = Utils::Fs::toUniformPath(p->new_name());
+    const Path oldFilePath = m_filePaths.at(fileIndex);
+    const auto newFilePath = Path(QString(p->new_name()));
 
     // Check if ".!qB" extension was just added or removed
-    if ((oldFilePath != newFilePath) && (oldFilePath != newFilePath.chopped(QB_EXT.size())))
+    // We should compare path in a case sensitive manner even on case insensitive
+    // platforms since it can be renamed by only changing case of some character(s)
+    if ((oldFilePath.data() != newFilePath.data())
+            && ((oldFilePath + QB_EXT) != newFilePath))
```

Comment:
you sure this logic is correct? I would stick with the previous logic to be safe.

Reply:
The logic is just adjusted to fit using new Path class.

Reply:
> you sure this logic is correct? I would stick with the previous logic to be safe.

In fact, it even fixes a bug in the old logic, since it explicitly uses `.!qB` extension, not just its length.